File: examples/local-agent/typescript/refund.mjs
/** Minimal application call through Auths and a generated domain client. */

import { connect } from "@auths-dev/sdk";
import { Stripe } from "@auths-dev/profile-stripe";

const session = await connect();
try {
  const stripe = new Stripe(session, { connection: "billing" });
  const refund = await stripe.refunds.create({
    paymentIntent: "pi_123",
    amount: 2_000,
    currency: "usd",
  });
  console.log(refund.id, refund.auths.operationId);
} finally {
  await session.close();
}

File: examples/local-agent/python/refund.py
"""Minimal application call through Auths and a generated domain client."""

import asyncio

import auths
from auths_profiles.stripe import Stripe


async def main() -> None:
    async with auths.connect() as session:
        stripe = Stripe(session, connection="billing")
        refund = await stripe.refunds.create(
            payment_intent="pi_123",
            amount=2_000,
            currency="usd",
        )
        print(refund.id, refund.auths.operation_id)


if __name__ == "__main__":
    asyncio.run(main())
